src/filters/simple-filter/filter-component.js

```diff
@@ -33,18 +32,28 @@ angular.module('patternfly.filters').component('pfFilter', {
         type: field.filterType,
         value: value
       };
-      if (!filterExists(newFilter)) {
+      var matchedFilter;
+
+      if (filterExists(newFilter)) {
 
         if (newFilter.type === 'select') {
           enforceSingleSelect(newFilter);
         }
 
+        if (field.filterType === 'complex-select' && !field.filterMultiselect && _.some(ctrl.config.appliedFilters, complexFilterExists)) {
+          _.remove(ctrl.config.appliedFilters, matchedFilter);
+        }
+
         ctrl.config.appliedFilters.push(newFilter);
 
         if (ctrl.config.onFilterChange) {
           ctrl.config.onFilterChange(ctrl.config.appliedFilters);
         }
       }
+
+      function complexFilterExists (item) {
+        return _.includes(item.value, _.split(value, field.filterDelimiter, 1)) ? matchedFilter = item : null;
+      }
```

Comment:
I agree with @jeff-phillips-18 , 'complexFilterExists' should just return a boolean, and 'matchedFilter' should be handled in a diff function, perhaps 'getComplexFilter'.  'complexFilterExists' just seems overloaded and difficult to 'grok' :-)  Pls consider breaking it up for better readability.  -thanks